### PR TITLE
[SIM_FLUTTER-19][FE] Create user dashboard page

### PR DESCRIPTION
**Pull request by RDLegs**
### Issue Links
- [Backlog](https://framgiaph.backlog.com/view/SIM_FLUTTER-19)
- [Slack](https://sun-philippine.slack.com/archives/C059L7XAXFB/p1686792140589019)

### Definition of Done
- [x] Created Dashboard Screen
- [x] Added Bottom Navigation Bar
- [x] Added Floating Action Button

### Notes
- Added mock data for account list

### Attachments
- Video Recording
https://github.com/framgia/sph-flutter/assets/99173155/67439d53-b968-4198-b42c-81c976f4b7cc

- Image
354433335_1400944540747803_349077909607380347_n

### CODE REVIEW

File: frontend/lib/main.dart [hunk 1]
```diff
@@ -30,33 +31,56 @@ class MyApp extends StatelessWidget {
...
         fontFamily: 'Nunito',
+        scaffoldBackgroundColor: const Color(0xFFDAEAEE),
         textTheme: const TextTheme(
           titleLarge: TextStyle(
+            color: Color(0xFF6D7881),
             fontSize: 30,
             fontWeight: FontWeight.bold,
           ),
           titleMedium: TextStyle(
+            color: Color(0xFF6D7881),
```
Comment (jstephend-sun) [inline frontend/lib/main.dart]: text theme color is repeated with the same values,

taken from sir egie's pr, has this implementation been tried?

image


maybe there are complications with that?

**Review by Dbulawan13 — DISMISSED** [review]
Approve PR #15 

Working:
- Able to create dashboard page
- The texts are rendered correctly.
- The navbar is rendered correctly based on the suggestions from Figma.
   - https://pub.dev/packages/curved_navigation_bar 

https://github.com/framgia/sph-flutter/assets/86758474/76158fe7-e9d6-414f-9896-65e35d61b109

Testing hours: 0.08
Comment: 0
Iteration: 0

File: frontend/lib/src/components/templates/header_image.dart [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import 'package:flutter/material.dart';
+
+/* HeaderImage is used as a design template for the image at the top left of all screens*/
```
Comment (jstephend-sun) [inline frontend/lib/src/components/templates/header_image.dart]: for consistency lets use multi line descriptor for the widget

File: frontend/lib/src/features/dashboard/components/account_card.dart [hunk 3]
```diff
@@ -0,0 +1,75 @@
+import 'package:flutter/material.dart';
+import 'package:frontend/src/features/dashboard/components/account_card_button.dart';
+import 'package:frontend/src/models/account.dart';
+
+class AccountCard extends StatelessWidget {
```
Comment (jstephend-sun) [inline frontend/lib/src/features/dashboard/components/account_card.dart:9]: add widget description here

File: frontend/lib/src/features/dashboard/components/account_card_button.dart [hunk 4]
```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+
+class AccountCardButton extends StatelessWidget {
```
Comment (jstephend-sun) [inline frontend/lib/src/features/dashboard/components/account_card_button.dart:6]: widget description

File: frontend/lib/src/features/dashboard/dashboard.dart [hunk 5]
```diff
@@ -0,0 +1,93 @@
+import 'package:flutter/material.dart';
+import 'package:frontend/src/features/dashboard/components/account_card.dart';
+import 'package:frontend/src/models/account.dart';
+
+class Dashboard extends StatelessWidget {
```
Comment (jstephend-sun) [inline frontend/lib/src/features/dashboard/dashboard.dart:12]: widget description

File: frontend/lib/src/features/dashboard/dashboard.dart [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+  Dashboard({
+    super.key,
+  });
+
+  /* This is the list of accounts pulled from BE */
+  /* NOTE: This is currently using mock data */
```
Comment (jstephend-sun) [inline frontend/lib/src/features/dashboard/dashboard.dart]:
```suggestion
  /* TODO (redempto): This is currently using mock data */
```

to find todos easier
you can also take a look at [todo tree ext](https://marketplace.visualstudio.com/items?itemName=Gruntfuggly.todo-tree)

**Review by jstephend-sun — CHANGES_REQUESTED** [review]
Requested Changes #15

Review time: 15mins
Comments: 4
Iteration: 1st review

File: frontend/lib/main.dart (hunk 1 above)
Comment (RDLegs) [reply frontend/lib/main.dart]: Ah, I was looking for this. This is a better implementation. Will change accordingly.

**Review by jstephend-sun — APPROVED** [review]
Approved #15

Review time: 15mins
Comments: 4
Iteration: 1st review

**Review by Dbulawan13 — APPROVED** [review]
Approve PR https://github.com/framgia/sph-flutter/pull/15

Working:
- Able to create dashboard page
- The texts are rendered correctly.
- The navbar is rendered correctly based on the suggestions from Figma.
    - https://pub.dev/packages/curved_navigation_bar
- When scrolling through the accounts, the plus (+) button will now appear behind them.

https://github.com/framgia/sph-flutter/assets/86758474/b6b7e84b-41c7-4896-9989-88f357d6673b

Testing Hours: 0.15
Comment: 0
Iteration: 0

**Pull request merged**